examples/example_local_search_evolution_strategies.py

```diff
@@ -0,0 +1,180 @@
+regression task that involves numeric constants. Local search via
+evolution strategies is used to determine numeric leaf values of the
+graph.
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+  Usage:
+    example_local_search_evolution_strategies.py [--max-generations=<N>]
+
+  Options:
+    -h --help
+    --max-generations=<N>  Maximum number of generations [default: 500]
+
+"""
+
+import functools
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.constants
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+# %%
+# We first define the target function. Note that this function contains
+# numeric values which are initially not available as constants to the search.
+
+
+def f_target(x):
+    return np.e * x[:, 0] ** 2 + 1.0 + np.pi
+
+
+# %%
+# Then we define the objective function for the evolution. It consists
+# of an inner objective which accepts a NumPy-compatible function as
+# its first argument and returns the mean-squared error between the
+# expression represented by a given individual and the target function
+# evaluated on a set of random points. This inner objective is used by
+# the local search to determine appropriate values for Parameter node
+# and the actual objective function to update the fitness of the
+# individual.
+
+
+def inner_objective(ind, seed):
+    """Return a loss for the numpy-compatible function f. Used for
+    calculating the fitness of each individual and for the local
+    search of numeric leaf values.
+
+    """
+
+    f = ind.to_numpy()
+    rng = np.random.RandomState(seed)
+    batch_size = 500
+    x = rng.uniform(-5, 5, size=(batch_size, 1))
+    y = f(x)
+    return -np.mean((f_target(x) - y[:, 0]) ** 2)
+
+
+def objective(individual, seed):
+    """Objective function of the regression task."""
+
+    if not individual.fitness_is_None():
+        return individual
+
+    individual.fitness = inner_objective(individual, seed)
+
+    return individual
+
+
+# %%
+# Next, we define the parameters for the population, the genome of
+# individuals, the evolutionary algorithm, and the local search.
+
+
+population_params = {"n_parents": 1, "seed": 818821}
+
+genome_params = {
+    "n_inputs": 1,
+    "n_outputs": 1,
+    "n_columns": 36,
+    "n_rows": 1,
+    "levels_back": None,
+    "primitives": (cgp.Add, cgp.Sub, cgp.Mul, cgp.Parameter),
+}
+
+ea_params = {
+    "n_offsprings": 4,
+    "mutation_rate": 0.05,
+    "tournament_size": 1,
+    "n_processes": 1,
+    "k_local_search": 2,
+}
+
+evolve_params = {"max_generations": int(args["--max-generations"]), "min_fitness": 0.0}
+
+# restrict the number of steps in the local search; since parameter
+# values are propagated from parents to offsprings, parameter values
+# may be iteratively improved across generations despite the small
+# number of steps per generation
+local_search_params = {"max_generations": 5}
```

Comment:
Just thinking, whether we maybe shouldn't name an iteration in the local search `generation` This might lead to confusion, especially since (correct me if i am wrong) 5 (or n) generations of local search are taken care of within one generation of cgp evolution.

Reply:
good point, maybe `step` is better than `generation`?

Reply:
done.

Reply:
sorry I forgot to reply